### PR TITLE
Deere new knobs

### PR DESCRIPTION
**Pull request by ferranlala**
I've made a new set of knobs for Deere that I think suit better its style. I've needed to add a bit of code to WKnobComposed though.

### CODE REVIEW

**Comment by Be-ing** [comment]
The white notches blend together with the white knob background for the EQs. The notch should be a different color.

**Comment by Be-ing** [comment]
Ooooooo I got a cool idea: make the EQ knob notches color coded to match the new RGB waveforms.

**Comment by ferranpujolcamins** [comment]
In fact my personal opinion is that all the knobs should have the notch
blue,

2015-08-02 22:22 GMT+02:00 Be-ing <notifications@github.com>:

**Comment by ferranpujolcamins** [comment]
In fact my personal opinion is that all the knobs should have the notch
blue,

2015-08-02 22:39 GMT+02:00 Ferran Pujol Camins <ferranpujolcamins@gmail.com>
:

File: src/widget/wknobcomposed.cpp [hunk 1]
```diff
@@ -38,6 +40,14 @@ void WKnobComposed::setup(QDomNode node, const SkinContext& context) {
     if (context.hasNode(node, "MaxAngle")) {
         m_dMaxAngle = context.selectDouble(node, "MaxAngle");
     }
+
+    if (context.hasNode(node, "RotationOffsetX")) {
```
Comment (daschuer) [inline src/widget/wknobcomposed.cpp]: We should keep the neames consistent with your Ring PR. 
There we use currently "RingCenterXOffset", so this schould become "KnobCenterXOffset".

**Comment by daschuer** [comment]
Other than the naming issue it LGTM. 

###

My personal Taste hints:
* The knobs are looking highlighted compared to the buttons. IMO they should look similar important. (until we have a idea which of them are more importent)
* It is Hard to Track that the Mix and Super knob when collapsing the effects. They are changing size shape and position.
* Now the sliders are looking foreign. At least the scale is overdone.

**Comment by ferranpujolcamins** [comment]
> The knobs are looking highlighted compared to the buttons. IMO they should look similar important. (until we have a idea which of them are more importent)

I think we should make the color of the buttons text the same color than the knobs. The contrast between what's important and what's background is too low for me.

> It is Hard to Track that the Mix and Super knob when collapsing the effects. They are changing size shape and position.

That was already a problem. Any ideas?

> Now the sliders are looking foreign. At least the scale is overdone.

They are going to be shrunk to make room for the VU meters, I'm preparing the corresponding PR. The next step will be to improve their looks. I'm starting to like this skin named Deere, I hope LateNight doesn't get jealous :)

**Comment by ferranpujolcamins** [comment]
By the way @daschuer , we have a tie with @Be-ing to decide how the knob notches should be colored. What's your take on this?

**Comment by daschuer** [comment]
> That was already a problem. Any ideas? 

A band aid solution would be to just keep Super and Mix in the Chain header. 
However the tiny knobs are kind of weak for introducing a ring later. 
So what about designing a double height header bar, including the channel knobs.  

> By the way @daschuer , we have a tie with @Be-ing to decide how the knob notches should be colored. What's your take on this?

The current EQ knobs are really looking almost "grayed out". 

Color coded knobs are very common on Studio Mixers and they are helpful to find the desired knob on a big grid. You can also find them on 4 channel DJ mixers. But are they beautiful? 

Does anyone know if there is a quasi standard for such colors? As far as I know EQ bands are not individual colored. 

If we ditch the different colors, we need a different kind of grouping for the EQ Knobs.  

Since this is a design question and  to much opinions in a design discussions are likely destroying the overall beauty of a design, Ferran should have th final word.

**Comment by Be-ing** [comment]
> Does anyone know if there is a quasi standard for such colors? As far as I know EQ bands are not individual colored. 

Yeah, they are with the RGB waveforms. Red is low, green is mid, blue is high.

**Comment by ferranpujolcamins** [comment]
>
> A band aid solution would be to just keep Super and Mix in the Chain
> header.
> However the tiny knobs are kind of weak for introducing a ring later.
> So what about designing a double height header bar, including the channel
> knobs.


What if we replace all the knobs in the fx chain with horizontal sliders?

2015-08-03 16:25 GMT+02:00 Be-ing <notifications@github.com>:

**Comment by daschuer** [comment]
> What if we replace all the knobs in the fx chain with horizontal sliders?

If you can effort the space ... sliders are a bit uncommon to tweak effect parameter. 
But if it looks beautiful, go for it. 

The resizing of the effect Rack is already annoying.  You should allow to display 8 parameter. 

Why 8: 
We currently already have the issue that we do not have the space for 8 Parameter, which was the compromise I have traded with Nicu during developing the LV2 branch. 
8 allows to tweak all "important" parameters of external LV2 effects, where we cannot control the amount of knobs. For effects with higher parameters it is likely that they are require their own GUI. 
But we allow also to select a set of 8 Parameter from many more parameters of the original LV2 effects to get around this.  

This might be a good solution: 
* design 4 small stick sized effect Racks with only the buttons required for performance
 (at least Super / Mix / Enable ....  and what else is required. 
* design a collapsible Setup area below, that offers all missing knobs but only for one effect at a time. This way you have the space to document the effect in full glance with Parameter Names units and values and Super knob mappings. We may also allow all possible effect hooks. 
All this is already part of the effect engine, but hidden. 
This would give Deere special effect nature among the other skins.

**Comment by ywwg** [comment]
> What if we replace all the knobs in the fx chain with horizontal sliders?

Let's not change things too much.  The fx in 1.12 aren't great, and it's too easy to make them overloaded, but we're not going to be able to fix that for this release.  Let's do the best we can for now.  Remember, the fx in 1.12 were always supposed to be super simple -- even having chains at all was out of scope.

Slightly changing the knob style is still ok for 1.12, although I haven't looked at this particular change.

**Comment by ferranpujolcamins** [comment]
I've changed the colours. Some of them are a bit arbitrary, any suggestions are welcome.

I must admit it looks cool. Shall we rename Deere to Rainbow Deer? :)

PS: I'll handle the effects knobs when we rearrange the fx section. Can this be merged anyway?

**Comment by Be-ing** [comment]
I like the EQs. However, I think the color coding of the EQs is lost in the randomness of the other knob notch colors. I think the EQs should be RGB and the rest should be one color that is neither red, green, nor blue. I don't know what that color should be though.

**Comment by ferranpujolcamins** [comment]
Like this?
captura de pantalla de 2015-08-04 08 49 36

**Comment by Be-ing** [comment]
Better, but it might be even better if there was a bit more of a difference between the high EQ knob and the rest of the knobs.

A bit off the topic of this PR, I noticed that the bottoms of the knobs are getting cut off in the effects.

**Comment by ferranpujolcamins** [comment]
> Better, but it might be even better if there was a bit more of a difference between the high EQ knob and the rest of the knobs.

Done

> This is a bit off the topic of this PR, but I notice that the bottoms of the knobs are getting cut off in the effects.

Yeah today I'm going to do a Deere FX refactoring PR to solve all this issues.

**Comment by daschuer** [comment]
Now we have lost the visual group of the EQs. Maybe all EQ bands should have the same color like this: 

https://images6.static-thomann.de/pics/bdb/361606/10016458_800.jpg
http://www.envirel.de/sirus-pro-mischpult-studio-i8/?gclid=CMDi_bbvjscCFSgGwwodAMIB-w
http://www.bax-shop.de/yamaha-mg16-mischpult-16-kanale

Watched some studio Mixers, and there seams to be relay a secret color code: 
Gain = Red  
EQ = Green/Blue

By the way, I have fond no R/G/B colored EQ knobs.

**Comment by Be-ing** [comment]
I don't think live mixer color conventions are relevant to a DJ program. All DJ equipment I've seen comes with one knob color, which is usually black with a white or off-white notch. If I had a controller or mixer with conventional knobs (mine has infinity encoders surrounded by LED rings and the caps don't come off), I'd get custom RGB knob caps.

**Comment by ferranpujolcamins** [comment]
Ok, we don't need to follow studio mixers colour code, no one will ever notice. We must prime usability. I certainly prefer all eq knobs with the same colour. 

This is my final take. I don't really care a bout the colours, I just want to move to other things. So if the only thing keeping this from merging are colours, please merge this and feel free to change them yourself :)

(Let me fine tune the small knobs on the fx units in a fx refactor PR I'm preparing)

**Comment by Be-ing** [comment]
Okay, I can play around with knob colors in all of the skins after this is merge.

**Comment by daschuer** [comment]
Currently the head Mix knob is not centered and the effect prepare region is gone.
I will merge this tonight.

**Comment by ferranpujolcamins** [comment]
Let me check that, I must messed with git.

2015-08-04 9:45 GMT+02:00 Daniel Schürmann <notifications@github.com>:

**Comment by ferranpujolcamins** [comment]
Fixed, sorry for the mistake.

> Currently the head Mix knob is not centered
I don't see nothing strange here. What do you mean? Can you send a screenshot?

**Comment by daschuer** [comment]
knob
All knobs are effected when turned to Min and Max. Center position is OK.

**Comment by ferranpujolcamins** [comment]
This doesn't happen to me. It is exactly what the new "KnobCenterXOffset" and "KnobCenterYOffset" should address. Maybe you forgot to rebuild from your previous checkout?

**Comment by daschuer** [comment]
Now it is working. My fault. 
Thank you for all the love you put into Deere!

**Comment by chtfn** [comment]
Just as a late note, after reading the comments: maybe basic RGB is not the best set of colours to go with as a general rule if we want to cater for colour-blind users. Not sure if this can be used as a reference somewhere in the documentation, but [here](http://www.cookbook-r.com/Graphs/Colors_%28ggplot2%29/#a-colorblind-friendly-palette) is a colorblind-friendly palette, which is based on [this paper](http://jfly.iam.u-tokyo.ac.jp/color/).

**Comment by Be-ing** [comment]
Thanks for that link. I'll look into it.

**Pull request merged**